R/mash_wrapper.R

```diff
@@ -416,70 +416,99 @@ merge_susie_cs <- function(susie_fit, coverage = "cs_coverage_0.95", complementa
 
 #' @importFrom data.table as.data.table setnames
 #' @export
-load_multitrait_R_sumstat <- function(
-    susie_fit, sumstats_db, coverage = NULL, top_loci = FALSE,
-    filter_file = NULL, remove_any_missing = FALSE, max_rows_selected = 300, nan_remove = FALSE,
-    exclude_condition, complementary = FALSE) {
-  extract_data <- function(sumstats_db) {
-    bhat <- as.data.table(cbind(sumstats_db$variant_names, sumstats_db$sumstats$betahat))
-    sbhat <- as.data.table(cbind(sumstats_db$variant_names, sumstats_db$sumstats$sebetahat))
-    setnames(bhat, c("variants", "bhat"))
-    setnames(sbhat, c("variants", "sbhat"))
-    bhat[, bhat := as.numeric(bhat)]
-    sbhat[, sbhat := as.numeric(sbhat)]
-    list(bhat = bhat$bhat, sbhat = sbhat$sbhat, variants = bhat$variants)
-  }
-
+load_multitrait_R_sumstat <- function(susie_fit,sumstats_db, coverage = NULL, top_loci = FALSE, filter_file = NULL, remove_any_missing = TRUE, max_rows_selected = 300, nan_remove=FALSE, exclude_condition, ld_meta_file) {
```

Comment:
All required argument without default parameters should be moved to before all the arguments with default paramteters. Also as I suggested ld_meta_file should be default to NULL which means no allele QC needed. So the interface should be:

```
load_multitrait_R_sumstat <- function(susie_fit,sumstats_db, exclude_condition=NULL, ld_meta_file = NULL, coverage = NULL, top_loci = FALSE, filter_file = NULL, remove_any_missing = TRUE, max_rows_selected = 300, nan_remove=FALSE)
```

and later you will have 

```r
if (!is.null(ld_meta_file)) {}
if (!is.null(exclude_condition)) {}
```